crates/kitsune_p2p/kitsune_p2p/src/gossip/sharded_gossip.rs

```diff
@@ -194,50 +195,26 @@ impl ShardedGossip {
             bandwidth,
         });
 
-        let mut agent_list_by_local_agents = vec![];
-        let mut all_agents = vec![];
         let mut refresh_agent_list_timer = std::time::Instant::now();
 
         metric_task_instrumented(config.tracing_scope.clone(), {
             let this = this.clone();
 
             async move {
+                let mut agent_info_session = this.create_agent_info_session().await?;
+
                 let mut stats = Stats::reset();
                 while !this
                     .gossip
                     .closing
                     .load(std::sync::atomic::Ordering::Relaxed)
                 {
                     tokio::time::sleep(GOSSIP_LOOP_INTERVAL).await;
-                    this.run_one_iteration(
-                        agent_list_by_local_agents.clone(),
-                        all_agents.as_slice(),
-                    )
-                    .await;
+                    this.run_one_iteration(&mut agent_info_session).await;
                     this.stats(&mut stats);
 
                     if refresh_agent_list_timer.elapsed() > AGENT_LIST_FETCH_INTERVAL {
-                        agent_list_by_local_agents =
-                            match this.gossip.query_agents_by_local_agents().await {
-                                Ok(a) => a,
-                                Err(e) => {
-                                    tracing::error!(
-                                        "Failed to query for agents by local agents - {:?}",
-                                        e
-                                    );
-                                    vec![]
-                                }
-                            };
-                        all_agents =
-                            match store::all_agent_info(&this.gossip.host_api, &this.gossip.space)
-                                .await
-                            {
-                                Ok(a) => a,
-                                Err(e) => {
-                                    tracing::error!("Failed to query for all agents - {:?}", e);
-                                    vec![]
-                                }
-                            };
+                        agent_info_session = this.create_agent_info_session().await?;
```

Comment:
I know you didn't introduce this, so feel free to leave as-is, but we do the agent refresh at the end, and the sleep at the beginning, so our agents will always be at least 100ms out of date.

Reply:
Good spot, rearranged statements so that we
1. Refresh agent info if required
2. Run an iteration
3. Sleep